src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/GenerateCoverSheetService.java

```diff
@@ -92,8 +103,9 @@ private void prepareCoverSheet(CaseDetails caseDetails, String partyAddressCcdFi
     }
 
     private AddressFoundInCaseData checkAddress(Map<String, Object> caseData, String partyAddressCcdFieldName,
-                                                String solicitorAddressCcdFieldName) {
-        return addressLineOneAndPostCodeAreBothNotEmpty((Map) caseData.get(solicitorAddressCcdFieldName)) ? AddressFoundInCaseData.SOLICITOR
+                                                String solicitorAddressCcdFieldName, boolean isRepresentedByASolicitor) {
+        return isRepresentedByASolicitor && addressLineOneAndPostCodeAreBothNotEmpty((Map) caseData.get(solicitorAddressCcdFieldName))
+            ? AddressFoundInCaseData.SOLICITOR
```

Comment:
needed added due to a bug when non represented respondents and applicants could have solicitor addresses saved to their casedata if they were originally represented then had their contact details updated in the update contact details event to not represented.